plugins/arStorageServicePlugin/modules/arStorageServicePlugin/actions/baseAction.class.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class arStorageServicePluginBaseAction extends sfAction
+{
+  const STORAGE_SERVICE_PACKAGE_PATH = 'file';
+
+  public function execute($request)
+  {
+    $view = sfView::NONE;
+
+    try
+    {
+      $view = $this->process($request);
+    }
+    catch (QubitApi404Exception $e)
+    {
+      $this->response->setStatusCode(404, $e->getMessage());
+    }
+    catch (QubitApiNotAuthorizedException $e)
+    {
+      $this->response->setStatusCode(401);
+    }
+    catch (QubitApiForbiddenException $e)
+    {
+      $this->response->setStatusCode(403, $e->getMessage());
+    }
+    catch (QubitApiBadRequestException $e)
+    {
+      $this->response->setStatusCode(400, $e->getMessage());
+    }
+    catch (Exception $e)
+    {
+      $this->response->setStatusCode(500);
+
+      throw $e;
+    }
+
+    return $view;
+  }
+
+  public function process($request)
+  {
+    $method = strtoupper($request->getMethod());
+    if (!method_exists($this, $method))
+    {
+      return $this->forward404();
+    }
+
+    // Define function callable
+    $fnCallable = array($this, $method);
+    $fnParamaters = array($request);
+
+    // Modern frameworks support application/json, Symfony1 is too old :)
+    // AngularJS doesn't use application/x-www-form-urlencoded
+    if (('PUT' == $method || 'POST' == $method) && 'application/json' == $request->getContentType())
+    {
+      $fnParamaters[] = json_decode($request->getContent());
```

Comment:
I think passing unsanitized parameters to `call_user_func_array()` is probably a insecure.

Reply:
I have removed this, but this same logic exists in the rest API plugin in both AtoM and Binder and should be reviewed if this is the case.